psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -1098,23 +1098,27 @@ void HF::guess() {
         // by Superposition of Atomic Densities", J Comput Chem 27,
         // 926 (2006).
 
-        // Build SAD density matrix
+        // Build non-idempotent, spin-restricted SAD density matrix
         compute_SAD_guess();
 
-        // Form initial spin-restricted Fock matrix from the SAD density
+        // Form initial (spin-restricted) Fock matrix from the SAD density.
         form_G();
         form_initial_F();
 
         // Diagonalize spin-restricted Fock matrix to get the same
         // orbitals for alpha and beta, also allowing spin-restricted
-        // open shell calculations to work
+        // open shell calculations to work.
         form_initial_C();
 
-        // Now can form the target density matrix
+        // Now can form the target density matrix, which may also
+        // correspond to a different charge and/or spin state than the
+        // neutral spin-restricted SAD guess. The density matrix is
+        // now also idempotent, so the energies one gets for the first
+        // printed iteration are variational.
         form_D();
 
         // This is a guess iteration: orbital occupations corresponded
-        // to SAD and must be reset in SCF
+        // to SAD and must be reset in SCF.
         iteration_ = -1;
```

Comment:
Note here where we mark the iteration as -1 which is later incremented to zero.

Reply:
... because if you put 0 then the answer will be wrong since it's not going to reset the occupations. I did try it at the outset!